scripts/polygon-eth/token_transfer.py

```diff
@@ -0,0 +1,316 @@
+                    'nonce': nonce
+                })
+
+                print("Processing input data from L2 to L1 tx (Polygonscan):", tx_hash)
+
+                (tx_hash, status) = send_tx("Root ERC20 contract to process tokens received on L1 tx (Etherscan):", tx, w3)
+                if (status == 1):
+                    return True
+                else:
+                    print("Receive transaction failed on L1 tx (Etherscan)", tx_hash)
+                    # Exit on tx error if it is a single L2 to L1 transfer in a while loop, otherwise just skip
+                    if exit_on_error:
+                        sys.exit(1)
+            except:
+                # Skip already processed tx
+                print("Payload processing on L1 failed")
+                return False
+        return False
+    except:
+        print(f"Proof generator response API timeout with error: {requests.exceptions}")
+        return False
+
+def receive_all_messages_l1(fx_erc20_root_tunnel_contract, fx_erc20_child_tunnel_contract, w3):
+    # Get last settled epoch event
+    event_filter = fx_erc20_child_tunnel_contract.events.MessageSent.create_filter(fromBlock=from_block_l2, toBlock="latest")
+    # Get all entries
+    entries = event_filter.get_all_entries()
+
+    # Traverse all entries
+    for entry in entries:
+        tx_hash = "0x" + binascii.hexlify(entry['transactionHash']).decode("utf-8")
+        receive_message_l1(tx_hash, fx_erc20_root_tunnel_contract, w3, False)
+
+
+# Distinguish between ledger or regular EOA
+if ledger:
+    w3_l1.middleware_onion.add(LedgerSignerMiddleware, "ledgereth_middleware")
+    w3_l2.middleware_onion.add(LedgerSignerMiddleware, "ledgereth_middleware")
+    accounts = get_accounts()
+    account = accounts[0]
+else:
+    account = Account.from_key(PRIVATE_KEY)
+    pk_hex = bytearray.fromhex(PRIVATE_KEY)
+account_address = account.address
+
+# Get bridge contracts
+# lp_token
+abi = get_abi(config['abi_lp_token_address'], config['url_api_polygon'], POLYGONSCAN_API_KEY)
+lp_token_contract = w3_l2.eth.contract(address=config['lp_token_address'], abi=abi)
+
+# bridged_erc20
+abi = get_abi(config['abi_bridged_erc20_address'], config['url_api_eth'], ETHERSCAN_API_KEY)
+bridged_erc20_contract = w3_l1.eth.contract(address=config['bridged_erc20_address'], abi=abi)
+
+# fx_erc20_root_tunnel
+abi = get_abi(config['abi_fx_erc20_root_tunnel_address'], config['url_api_eth'], ETHERSCAN_API_KEY)
+fx_erc20_root_tunnel_contract = w3_l1.eth.contract(address=config['fx_erc20_root_tunnel_address'], abi=abi)
+
+# fx_erc20_root_tunnel
+abi = get_abi(config['abi_fx_erc20_child_tunnel_address'], config['url_api_polygon'], POLYGONSCAN_API_KEY)
+fx_erc20_child_tunnel_contract = w3_l2.eth.contract(address=config['fx_erc20_child_tunnel_address'], abi=abi)
+
+operation = ""
+amount = 0
+destination = ""
+# Parse command line arguments
+opts, args = getopt.getopt(sys.argv[1:], "ho:a:d:", ["operation=","amount=","destination="])
```

Comment:
Consider the cli tooling Click
https://click.palletsprojects.com/en/8.1.x/

Will save you a lot of the parsing stuff and you get validation for free etc :)

Reply:
We'll do it in on of the following PRs